### ISSUE TITLE
[GCC13] Audacity 3.2.5 => error:  nyqstk/include/Stk.h:360:18: error: declaration does not declare anything [-fpermissive]

### ISSUE THREAD

**Issue by admnd** [audacity/audacity#4489]
### Bug description

The compilation fails with the following error:

```
FAILED: lib-src/libnyquist/CMakeFiles/libnyquist.dir/nyquist/nyqstk/stkinit.cpp.o 
/var/tmp/portage/media-sound/audacity-3.2.5/work/audacity-Audacity-3.2.5/lib-src/libnyquist/nyquist/nyqstk/include/Stk.h:360:18: error: declaration does not declare anything [-fpermissive]
```

### Steps to reproduce

Simply => ``emerge '=media-sound/audacity-3.2.5'``on a Gentoo box that uses GCC 13.

### Expected behavior

The code should compile.

### Audacity Version

stable version (from a Linux repository)

### Operating system

Linux

**Comment by FalCT60** [comment]
AFAIK, GCC13 is reported as faulty in lots of other sw compilations.
I could compile master on my system, but I am still using GCC11.

**Comment by admnd** [comment]
> AFAIK, GCC13 is reported as faulty in lots of other sw compilations. I could compile master on my system, but I am still using GCC11.

GCC 13 is not the problem but the code of applications IS. A compiler is like wine, the more the years passes the more respectful to standards it becomes (so "clunky" code and tinkering are less tolerated), not counting its own bugs fixed. 

In the present issue, the fix is this one:

```
--- a/lib-src/libnyquist/nyquist/nyqstk/include/Stk.h	2023-04-04 08:28:25.060827847 -0400
+++ b/lib-src/libnyquist/nyquist/nyqstk/include/Stk.h	2023-04-04 08:30:22.174162825 -0400
@@ -47,6 +47,7 @@
 #endif
 
 #include <stddef.h>
+#include <stdint.h>
 #include <string>
 #include <iostream>
 #include <sstream>
@@ -355,10 +356,6 @@
 typedef signed int SINT32;
 typedef float FLOAT32;
 typedef double FLOAT64;
-// added by RBD
-#ifdef __linux__
-#define intptr_t long
-#endif
 
 // The default sampling rate.
 const StkFloat SRATE = 44100.0;

```

The code is what it is but "manually" redefining existing standard types is not the thing to do. Bunch of troubles ahead especially using a **#define** rather than a **typedef**.... I don't have a Windows machine w/ Visual C++ but stdint.h is not Linux specific.

Originally reported here => https://bugs.gentoo.org/899764

**Comment by crsib** [comment]
Well, you are correct that `#include <stdint.h>` is a, probably, a better way to get `intptr_t`. 

However,

1. `intptr_t` is not required by C standard,
2. is it wise to break the old code? :-) 

I expect that your patch will compile on all compilers Audacity supports, so it is more about the philosophy and aging wine.

We will investigate issues with GCC13 when it lands in Arch or Fedora. You are welcome submitting patches before that, in case the regular Audacity CI passes.

**Comment by admnd** [comment]
Old code is already broken so... time to fix it :)

Perhaps a "safer" way is to enclose **#include <stdint.h>** inside a conditional **#ifdef __linux__ .... #endif**. 

I let you judge what the best option is, you are definitely more familiar than me with the source code. The above suggestion is a quick-fix that worked for me on Gentoo Linux. 

PS: Fedora 38 will include GCC 13 AFAIK.

**Comment by crsib** [comment]
This is now fixed in `release-3.3.0`, fix is expected in master on Monday Apr 24
